lib/features/children/generosity_challenge_chat/chat_scripts/utils/chat_script_registration_handler.dart

```diff
@@ -31,20 +31,48 @@ class ChatScriptRegistrationHandler {
 
   Future<bool> _registerUser() async {
     final userData = _generosityChallengeRepository.loadUserData();
-    log(_sharedPreferences.getKeys().toString());
+    var email = '';
+    try {
+      email = _sharedPreferences.getString(ChatScriptSaveKey.email.value) ?? '';
+    } catch (e) {
+      email = userData[ChatScriptSaveKey.email.value] as String;
+      if (email.isEmpty) {
+        LoggingInfo.instance.error(
+          'Failed to get sign up email: $e',
+          methodName: '_registerUser',
+        );
+      }
+    }
+    if (email.isEmpty) {
+      return false;
+    }
+
     final firstname = userData[ChatScriptSaveKey.firstName.value] as String;
     final lastname = userData[ChatScriptSaveKey.lastName.value] as String;
-    final email = userData[ChatScriptSaveKey.email.value] as String;
     final password = userData[ChatScriptSaveKey.password.value] as String;
+    final phoneNr = (userData[ChatScriptSaveKey.phone.value] ??
+        Util.defaultUSPhoneNumber) as String;
     final result = await _authRepository.registerGenerosityChallengeUser(
       firstname: firstname,
       lastname: lastname,
       email: email,
       password: password,
+      phoneNumber: phoneNr,
     );
-    await _generosityChallengeRepository.setAlreadyRegistered(
-      isAlreadyRegistered: result.isAlreadyRegistered,
-    );
+    // if the result is success and they are not already registered
```

Comment:
Maybe i'm going cuckoo, but the outcome of 

`await _generosityChallengeRepository.setAlreadyRegistered(
      isAlreadyRegistered: result.isAlreadyRegistered,
    );`
    
    And the new code does exactly the same but more convoluted?

Reply:
no because this will now be triggered twice. Once on day one and once on day 7.
In the previous implementation was used we would create a user on day 1, set already registered to true, and kick them out on day 8.